src/app/payments/update-pending-payments.ts

```diff
@@ -240,31 +249,44 @@ const updatePendingPayment = wrapAsyncToRunInSpan({
 const reconstructPendingPaymentFlow = async <
   S extends WalletCurrency,
   R extends WalletCurrency,
->(
-  paymentHash: PaymentHash,
-): Promise<PaymentFlow<S, R> | ApplicationError> => {
+>({
+  paymentHash,
+  inputAmount,
+}: {
+  paymentHash: PaymentHash
+  inputAmount: number
+}): Promise<PaymentFlow<S, R> | ApplicationError> => {
   const ledgerTxns = await LedgerService().getTransactionsByHash(paymentHash)
   if (ledgerTxns instanceof Error) return ledgerTxns
 
   const nonEndUserWalletIds = Object.values(await getNonEndUserWalletIds())
 
-  const payment = ledgerTxns.find(
+  const payments = ledgerTxns.filter(
     (tx) =>
       tx.pendingConfirmation === true &&
       tx.type === LedgerTransactionType.Payment &&
       tx.debit > 0 &&
       tx.walletId !== undefined &&
       !nonEndUserWalletIds.includes(tx.walletId),
-  ) as LedgerTransaction<S> | undefined
-  if (!payment) return new CouldNotFindTransactionError()
+  ) as LedgerTransaction<S>[] | undefined
+  if (!(payments && payments.length)) return new CouldNotFindTransactionError()
 
-  const { walletId: senderWalletId } = payment
+  const { walletId: senderWalletId } = payments[0]
   if (!senderWalletId) return new MissingPropsInTransactionForPaymentFlowError()
   const senderWallet = await WalletsRepository().findById(senderWalletId)
   if (senderWallet instanceof Error) return senderWallet
   const senderAccount = await AccountsRepository().findById(senderWallet.accountId)
   if (senderAccount instanceof Error) return senderAccount
 
+  const filteredPayments = payments.filter((tx) =>
+    senderWallet.currency === WalletCurrency.Btc
+      ? tx.satsAmount === inputAmount
+      : tx.centsAmount === inputAmount,
+  )
+  if (!(filteredPayments && filteredPayments.length))
+    return new CouldNotFindTransactionError()
```

Comment:
not sure about this.... so I cant re-try a failed invoice? an example can be done with OBW, create an invoice, close the app, try a payment (it will fail because is not connected) then open OBW wait to connect and try payment again (should succeed)

Reply:
I don't understand how this affects retry. As-is now now can always retry as tested here? https://github.com/GaloyMoney/galoy/pull/2151/files#diff-15e9812156b4ff1c33faaaf99ffcc368a8af2756ccb7a5e81acdcfc46782db9fR2363